internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -324,12 +324,13 @@ type image struct {
 }
 
 type Input struct {
-	Attestations []attestationData `json:"attestations"`
-	Image        image             `json:"image"`
+	Attestations  []attestationData       `json:"attestations"`
+	Image         image                   `json:"image"`
+	AllComponents []app.SnapshotComponent `json:"allcomponents"`
 }
 
 // WriteInputFile writes the JSON from the attestations to input.json in a random temp dir
-func (a *ApplicationSnapshotImage) WriteInputFile(ctx context.Context) (string, []byte, error) {
+func (a *ApplicationSnapshotImage) WriteInputFile(ctx context.Context, allcomp []app.SnapshotComponent) (string, []byte, error) {
```

Comment:
Rather that doing this, hold the `*app.SnapshotSpec` in the `ApplicationSnapshotImage` struct